include/revng/Model/Pass/PurgeUnnamedAndUnreachableTypes.h

```diff
@@ -9,7 +9,9 @@
 namespace model {
 
 /// Remove all the types that cannot be reached from any named type or a type
-/// "outside" the type system itself.
+/// "outside" the type system itself. If `pruneAllUnusedTypes` is set to true
+/// remove all the types that cannot be reached from any type from `Functions`.
+template<bool pruneAllUnusedTypes>
```

Comment:
* `PruneAllUnusedTypes`, does this pass check conventions?
* Can you make this a private `...Impl` function and introduce two new functions with a sensible name? I'm not a fan of `bool` arguments.